hkmc2/shared/src/main/scala/hkmc2/semantics/Term.scala

```diff
@@ -35,10 +36,12 @@ enum Term extends Statement:
   case Ret(result: Term)
   case Try(body: Term, finallyDo: Term)
   case Handle(lhs: LocalSymbol, rhs: Term, defs: ObjBody)
+  case Mod(inner: Term)
```

Comment:
Why do we need a new term form? Why not use the usual `Ref` and `Sel` and check whether something is a module by looking at its symbol?

Reply:
This means a term is modified by a `module` modifier keyword. For example,

```fs
fun f: module M
fun g[T]: module T
```

Here the results get elaborated into

```
Mod(Sel(Ref(M)))
Mod(Ref(T))
```

So I need that to check if a result type is correctly marked as `module`.

Of course, it is not necessary for now that we can always parse the Tree, but I think it'd be useful in future because we will inevitably elaborate some `module` things.

Reply:
Ah ok it's only for module type annotations. Then the name should be more informative, like `ModType`. Though TBH I'm not sure we need a term for that. After elaboration, I suspect no one cares whether the user wrote `: module M` or `: M`. The `module` part is just for the elaborator to do its checks.

Reply:
I think you are probably right that we may not need this. I'll try to remove it and see if anything stops us from doing that.

Reply:
removed!